language/tools/move-mv-llvm-compiler/src/stackless/llvm.rs

```diff
@@ -317,6 +317,88 @@ impl Builder {
         }
     }
 
+    /// Load a struct pointer alloca, add a field offset to it, and store the new pointer value.
+    pub fn field_ref_store(&self, src: Alloca, dst: Alloca, struct_ty: StructType, offset: usize) {
+        unsafe {
+            let ty = src.llvm_type().0;
+            let tmp_reg = LLVMBuildLoad2(self.0, ty, src.0, "tmp".cstr());
+            let field_ptr = LLVMBuildStructGEP2(
+                self.0,
+                struct_ty.0,
+                tmp_reg,
+                offset as libc::c_uint,
+                "fld_ref".cstr(),
+            );
+            LLVMBuildStore(self.0, field_ptr, dst.0);
+        }
+    }
+
+    // Load the source fields, insert them into a new struct value, then store the struct value.
+    pub fn insert_fields_and_store(
+        &self,
+        src: &[(Type, Alloca)],
+        dst: (Type, Alloca),
+        stype: StructType,
+    ) {
+        unsafe {
+            let loads = src
+                .iter()
+                .enumerate()
+                .map(|(i, (ty, val))| {
+                    let name = format!("fv.{i}");
+                    LLVMBuildLoad2(self.0, ty.0, val.0, name.cstr())
+                })
+                .collect::<Vec<_>>();
+
+            let mut agg_val = LLVMGetUndef(stype.0);
+            for (i, ld) in loads.iter().enumerate() {
+                let s = format!("insert_{i}").cstr();
+                agg_val = LLVMBuildInsertValue(self.0, agg_val, *ld, i as libc::c_uint, s);
+            }
+
+            assert_eq!(LLVMTypeOf(agg_val), dst.0 .0);
+            LLVMBuildStore(self.0, agg_val, dst.1 .0);
+        }
+    }
+
+    // Load the source struct, extract fields , then store each field in a local.
+    pub fn load_and_extract_fields(
+        &self,
+        src: (Type, Alloca),
+        dst: &[(Type, Alloca)],
+        stype: StructType,
+    ) {
+        unsafe {
+            //let tmp_reg = LLVMBuildLoad2(self.0, src.0.ptr_type().0, src.1.0, "tmp".cstr());
+            //let srcval = LLVMBuildLoad2(self.0, stype.0, tmp_reg, "srcval".cstr());
```

Comment:
probably overlooked commented out code?

Reply:
Thanks, I will delete that before commit.